api/src/main/java/org/apache/iceberg/TableScan.java

```diff
@@ -49,6 +49,24 @@
    */
   TableScan useSnapshot(long snapshotId);
 
+  /**
+   * Create a new {@link TableScan} from this scan's configuration that will use the given branch
+   *
+   * @param branch the branch to scan from
+   * @return a new scan based on this with the given branch
+   * @throws IllegalArgumentException if the branch cannot be found
+   */
+  TableScan branch(String branch);
```

Comment:
does the distinction between branch/tag really matter here or could we just have a `TableScan ref(String ref)` method?

Reply:
It's a good point. My thought was that from an API point of view people would be more familiar with the terms branch and tag, so they would leverage that. I also think if is using this, they know before hand if it's a branch or a tag, so it's not cumbersome. But yeah currently, the logic is effectively the same which is to use the snapshot for the reference.

Thoughts?

Reply:
Would `useRef` be better? Just to match `useSnapshot`.

Reply:
Actually I think there is a need to distinguish `useBranch` and `useTag`, because `useBranch` can be used in combiantion with `asOfTime` to perform time travel against a specific branch.